frontend/src/components/CreateEvent/CreateEvent.jsx

```diff
@@ -69,6 +93,27 @@ function CreateEvent ({onClose, onCreate, onEdit, initialView, isEdit}) {
                     <p>Date</p>
                     <input type="date" placeholder="mm/dd/yyyy" value={date} onChange={(e) => setDate(e.target.value)} required/>
 
+                    <p>Duration</p>
+                    <input type="number" value={duration} onChange={(e) => setDuration(e.target.value)} placeholder="Duration in 30-min slots" min="1" />
+                    <button type="button" onClick={fetchRecommendations}>Get Recommendations</button>
+                    {recommendations.length > 0 && (
+                        <div>
+                            <h2>Recommended Times</h2>
+                            <ul>
+                            {recommendations.map((rec, index) => {
+                                const startTime = slotToTime(rec);
+                                const endTime = slotToTime(rec + duration);
+                                return (
+                                    <li key={index} onClick={() => {
```

Comment:
nice, you made these clickable! I would suggest making these more obviously buttons, as it's not very clear they are right now.